src/host/warpcast/scraper.ts

```diff
@@ -64,8 +66,144 @@ export class Scraper {
+          return;
+        }
+        const userFullNameNode = userNode.querySelector(
+          ':scope > div > div > span',
+        );
+        const username = link.href.split('/').at(-1);
+
+        if (!userFullNameNode?.parentElement || !username) {
+          return;
+        }
+
+        const rect = userFullNameNode.getBoundingClientRect();
+
+        return {
+          node: userFullNameNode.parentElement,
+          top: rect.top + window.scrollY,
+          data: {
+            username,
+          },
+        };
+      })
+      .filter(Boolean);
+  }
+
+  private static getMain() {
+    return document.querySelector('main');
+  }
+
+  private static getHandleUser(): UserScrapingResult | undefined {
+    const pathname = window.location.pathname;
+    if (!isUserPathname(pathname)) {
+      return;
+    }
+    const username = extractUsernameFromPathname(pathname);
+    const main = Scraper.getMain();
+    if (!main || !username) {
+      return;
+    }
+    const allDivsOfMain = [...main.querySelectorAll('div')];
+    const usernameNode = allDivsOfMain.find((div) => {
+      return div.textContent === `@${username}`;
+    });
+    const userFullNameNode =
+      usernameNode?.parentElement?.firstElementChild?.firstElementChild;
+    if (!userFullNameNode?.parentElement) {
+      return;
+    }
+
+    const rect = userFullNameNode.getBoundingClientRect();
+
+    return {
+      node: userFullNameNode.parentElement,
+      top: rect.top + window.scrollY,
+      data: {
+        username,
+      },
+    };
+  }
+
   public static getUsers(): UserScrapingResult[] {
-    // not used for now
-    return [];
+    Scraper.getHandleUser();
+    const posts = Scraper.getPosts();
+    const usersFromPosts = posts
+      .map((post) => {
+        const links = [...post.node.querySelectorAll('a')];
+        const userFullName = links
+          .map((link) => {
+            const isUserLink = link.href.endsWith(post.data.authorUsername);
+            if (!isUserLink) {
+              return;
+            }
+            const userFullNameNode = link.querySelector(':scope > div > span');
+            if (!userFullNameNode?.parentElement) {
+              return;
+            }
+
+            const rect = userFullNameNode.getBoundingClientRect();
+
+            return {
+              node: userFullNameNode.parentElement,
+              top: rect.top + window.scrollY,
+              data: {
+                username: post.data.authorUsername,
+              },
+            };
+          })
+          .find(Boolean);
+
+        return userFullName;
+      })
+      .filter(Boolean);
+
+    const usersFromSuggestedFollow = Scraper.getSuggestedFollowsSectionUsers();
+    const handleUser = Scraper.getHandleUser();
+
+    if (handleUser) {
+      return [...usersFromPosts, ...usersFromSuggestedFollow, handleUser];
+    }
+
+    return [...usersFromPosts, ...usersFromSuggestedFollow];
   }
 }
```

Comment:
@SolutionsEngineer in this file I took a bit better (imo) approach for scraping - I made composition of well named sub-methods so it is easier to follow what is going on.

Imho it's much better than what we currently have on Twitter which is some crazy query selector that no one understands 😅

Reply:
Yeah, it was really nice to review that, and I think it will pay off if something changes in the future. It will be much easier to identify which function stops working. Great approach.